src/v/config/tls_config.h

```diff
@@ -80,75 +80,18 @@ class tls_config {
-    get_credentials_builder() const& {
-        if (_enabled) {
-            return ss::do_with(
-              ss::tls::credentials_builder{},
-              [this](ss::tls::credentials_builder& builder) {
-                  builder.set_priority_string("PERFORMANCE:%SERVER_PRECEDENCE");
-                  builder.set_dh_level(ss::tls::dh_params::level::MEDIUM);
-                  if (_require_client_auth) {
-                      builder.set_client_auth(ss::tls::client_auth::REQUIRE);
-                  }
-
-                  auto f = _truststore_file ? builder.set_x509_trust_file(
-                             *_truststore_file, ss::tls::x509_crt_format::PEM)
-                                            : builder.set_system_trust();
-
-                  if (_crl_file) {
-                      f = f.then([this, &builder] {
-                          return builder.set_x509_crl_file(
-                            *_crl_file, ss::tls::x509_crt_format::PEM);
-                      });
-                  }
-
-                  if (_key_cert) {
-                      f = f.then([this, &builder] {
-                          return builder.set_x509_key_file(
-                            (*_key_cert).cert_file,
-                            (*_key_cert).key_file,
-                            ss::tls::x509_crt_format::PEM);
-                      });
-                  }
-
-                  return f.then([&builder]() {
-                      return std::make_optional(std::move(builder));
-                  });
-              });
-        }
 
-        return ss::make_ready_future<
-          std::optional<ss::tls::credentials_builder>>(std::nullopt);
-    }
+    get_credentials_builder() const&;
 
     ss::future<std::optional<ss::tls::credentials_builder>>
-    get_credentials_builder() && {
-        auto ptr = ss::make_lw_shared(std::move(*this));
-        return ptr->get_credentials_builder().finally([ptr] {});
-    }
-
-    static std::optional<ss::sstring> validate(const tls_config& c) {
-        if (c.get_require_client_auth() && !c.get_truststore_file()) {
-            return "Trust store is required when client authentication is "
-                   "enabled";
-        }
+    get_credentials_builder() &&;
 
-        return std::nullopt;
-    }
+    static std::optional<ss::sstring> validate(const tls_config& c);
 
     bool operator==(const tls_config& rhs) const = default;
 
     friend std::ostream&
-    operator<<(std::ostream& o, const config::tls_config& c) {
-        o << "{ "
-          << "enabled: " << c.is_enabled() << " "
-          << "key/cert files: " << c.get_key_cert_files() << " "
-          << "ca file: " << c.get_truststore_file() << " "
-          << "crl file: " << c.get_crl_file() << " "
-          << "client_auth_required: " << c.get_require_client_auth() << ""
-          << " }";
-        return o;
-    }
+    operator<<(std::ostream& o, const config::tls_config& c);
```

Comment:
Why not also move the other implementation stuff?

Should be able to get includes down to:
```c++
#include "base/seastarx.h"

#include <seastar/core/future.hh>
#include <seastar/core/sstring.hh>
#include <seastar/net/tls.hh>

#include <yaml-cpp/node/node.h>

#include <iosfwd>
#include <optional>
```